oclc/classify.py

```diff
@@ -769,6 +951,72 @@ def lookup_by(self, **kwargs):
         representation, cached = Representation.get(self._db, url)
         return representation.content
 
+class OCLCLookupCoverageProvider(MetadataWranglerBibliographicCoverageProvider):
+
+    def __init__(self, collection, api=None, **kwargs):
+        super(OCLCLookupCoverageProvider, self).__init__(
+            collection, registered_only=False, **kwargs
+        )
+        self.api = api or OCLCClassifyAPI(self._db)
+
+
+class IdentifierLookupCoverageProvider(OCLCLookupCoverageProvider):
+    """Does identifier (specifically, ISBN) lookups using OCLC Classify.
+    """
+    SERVICE_NAME = "OCLC Classify Identifier Lookup"
+    INPUT_IDENTIFIER_TYPES = [Identifier.ISBN]
+    DATA_SOURCE_NAME = DataSource.OCLC
+
+    parser = OCLCClassifyXMLParser()
+
+    def process_item(self, identifier):
+        """Ask OCLC Classify about a single ISBN and create an Edition
+        based on what it says.
+        """
+        # Perform a title/author lookup.
+
+        metadata_list = []
+
+        try:
+            xml = self.api.lookup_by(isbn=identifier.identifier)
+            tree, code, owi_numbers = self.parser.initial_look_up(self._db, xml)
```

Comment:
Let's try calling `owi_numbers` something like `substitutes`, and have `initial_look_up` return the IdentifierData objects rather than the raw numbers.

That way we can understand `_single` as the 'we found the data' method, and `_multiple` as the 'we didn't find the data, but we found some equivalent items' method.

I think we can also avoid dealing with the `code` here -- we can have a rule that if `substitutes` is `None` then the document contains the data we want (so we call _single), and otherwise the document is useless except for the list of substitutes (so we should call _multiple).